internal/pkg/skuba/addons/kured_test.go

```diff
@@ -0,0 +1,161 @@
+			if got := GetKuredImage(tt.args.imageTag); got != tt.want {
+				t.Errorf("GetKuredImage() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func Test_renderContext_KuredImage(t *testing.T) {
+	type fields struct {
+		addon  Addon
+		config AddonConfiguration
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		want   string
+	}{
+		// TODO: Add test cases.
+		{
+			name: "Test_renderContext_KuredImage_1.15.2",
+			fields: fields{
+				addon: Addon{
+					addon:             kubernetes.Kured,
+					templater:         renderKuredTemplate,
+					callbacks:         nil,
+					addonPriority:     normalPriority,
+					getImageCallbacks: []getImageCallback{GetKuredImage},
+				},
+				config: AddonConfiguration{
+					ClusterVersion: version.MustParseSemantic("1.15.2"),
+					ControlPlane:   "",
+					ClusterName:    "",
+				},
+			},
+			want: "registry.suse.de/devel/caasp/4.0/containers/containers/caasp/v4/kured:[[:digit:]]{1,}.[[:digit:]]{1,}.[[:digit:]]{1,}[-rev[:digit:]{1,}]{0,1}",
+		},
+		{
+			name: "Test_renderContext_KuredImage_1.16.2",
+			fields: fields{
+				addon: Addon{
+					addon:             kubernetes.Kured,
+					templater:         renderKuredTemplate,
+					callbacks:         nil,
+					addonPriority:     normalPriority,
+					getImageCallbacks: []getImageCallback{GetKuredImage},
+				},
+				config: AddonConfiguration{
+					ClusterVersion: version.MustParseSemantic("1.16.2"),
+					ControlPlane:   "",
+					ClusterName:    "",
+				},
+			},
+			want: "registry.suse.de/devel/caasp/4.0/containers/containers/caasp/v4/kured:[[:digit:]]{1,}.[[:digit:]]{1,}.[[:digit:]]{1,}[-rev[:digit:]{1,}]{0,1}",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			renderContext := renderContext{
+				addon:  tt.fields.addon,
+				config: tt.fields.config,
+			}
+			got := renderContext.KuredImage()
+			matched, err := regexp.Match(tt.want, []byte(got))
+			if err != nil {
+				t.Errorf(err.Error())
+			}
+			if !matched {
+				t.Errorf("renderContext.KuredImagee() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func Test_renderKuredTemplate(t *testing.T) {
+	type args struct {
+		addonConfiguration AddonConfiguration
+	}
+	tests := []struct {
+		name string
+		args args
+		want string
+	}{
+		// TODO: Add test cases.
+		{
+			name: "Test_renderKuredTemplate_1.15.2",
+			args: args{
+				addonConfiguration: AddonConfiguration{
+					ClusterVersion: version.MustParseSemantic("1.15.2"),
+					ControlPlane:   "",
+					ClusterName:    "",
+				},
+			},
+			want: kuredManifest,
+		},
+		{
+			name: "Test_renderKuredTemplate_1.16.2",
+			args: args{
+				addonConfiguration: AddonConfiguration{
+					ClusterVersion: version.MustParseSemantic("1.16.2"),
+					ControlPlane:   "",
+					ClusterName:    "",
+				},
+			},
+			want: kuredManifest,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := renderKuredTemplate(tt.args.addonConfiguration); got != tt.want {
```

Comment:
Test `renderKuredTemplate` includes testing KuredImage(), therefore, the test case `Test_renderContext_KuredImage` could be drop off.

If me, I would like to test after rendering, is all the go template got rendered?
This could also cover the bug https://bugzilla.suse.com/show_bug.cgi?id=1141863

Reply:
agreed. removed.